pkg/volume/plugins.go

```diff
@@ -750,6 +750,9 @@ func (pm *VolumePluginMgr) refreshProbedPlugins() {
 
 // ListVolumePluginWithLimits returns plugins that have volume limits on nodes
 func (pm *VolumePluginMgr) ListVolumePluginWithLimits() []VolumePluginWithAttachLimits {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
```

Comment:
How about use only read lock ?

pm.mutex.RLock()
defer pm.mutex.RUnlock()

Reply:
Emm... But `sync.Mutex` only supports `Lock`. Maybe you thought it as `sync.RWMutex` ?
```go
// VolumePluginMgr tracks registered plugins.
type VolumePluginMgr struct {
	mutex         sync.Mutex
	plugins       map[string]VolumePlugin
	prober        DynamicPluginProber
	probedPlugins map[string]VolumePlugin
	Host          VolumeHost
}
```